pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -388,19 +397,65 @@ func (r *Reconciler) streamLogs(ctx context.Context, o results.Object, logType,
 		return fmt.Errorf("error reading from tkn reader: %w", err)
 	}
 
-	errChanRepeater := make(chan error)
-	go func(echan <-chan error, o metav1.Object) {
-		writeErr := <-echan
-		errChanRepeater <- writeErr
+	errChanRepeater := make(chan error, 100)
 
-		_, err := writer.Flush()
+	// logctx is derived from ctx. Therefore, if ctx is cancelled (either explicitly through a call to its cancel
+	// function or when it reaches its deadline), logctx will be cancelled automatically.
+	logctx, _ := context.WithTimeout(ctx, 10*time.Minute)
```

Comment:
put a TODO comment about we are deferring on making this configurable for now until feedback from users uncovers the need for it

Reply:
done